pkgs/stdenv/booter.nix

```diff
@@ -98,8 +98,8 @@ stageFuns: let
     thisStage =
       if args.__raw or false
       then args'
-      else allPackages ((builtins.removeAttrs args' ["selfBuild"]) // {
-        actuallySplice = !(args.selfBuild or true);
+      else allPackages (args' // {
+        actuallySplice = thisStage.stdenv.buildPlatform != thisStage.stdenv.targetPlatform;
```

Comment:
No need to check hostPlatform too?

Reply:
Nixpkgs doesn't allow both `build!=host` and `host!=target` at the same time.  So if `build==target` then it must be true that `build==host==target`.

But I guess that might not be true in the future so please see bfc7413cfe6ec